src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs

```diff
@@ -635,9 +636,47 @@ private TestMethodAttribute GetTestMethodAttribute(MethodInfo methodInfo, TestCl
         /// <returns> The <see cref="MethodInfo"/>. </returns>
         private MethodInfo GetMethodInfoForTestMethod(TestMethod testMethod, TestClassInfo testClassInfo)
         {
-            var methodsInClass = testClassInfo.ClassType.GetRuntimeMethods().ToArray();
             MethodInfo testMethodInfo;
 
+            if (testMethod.HasManagedMethodAndType)
+            {
+                var assembly = testClassInfo.ClassType.GetTypeInfo().Assembly;
+                var methodBase = ManagedNameHelper.GetMethod(assembly, testMethod.ManagedTypeName, testMethod.ManagedMethodName);
+
+                if (methodBase is MethodInfo mi)
+                {
+                    testMethodInfo = mi;
+                }
+                else
+                {
+                    var parameters = methodBase.GetParameters().Select(i => i.ParameterType).ToArray();
+                    testMethodInfo = methodBase.DeclaringType.GetRuntimeMethod(methodBase.Name, parameters);
```

Comment:
Could DeclaringType be null?

Reply:
Good question. This was a dirty hack to convert MethodBase into MethodInfo. Any cleaner recommendations? 

I’ll check if that could be null.

Reply:
In the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.declaringtype?view=net-5.0#remarks) it is stated that `DeclaredType` is null only if the member is `global`. Since this method is always inside a test class it shouldn't be null.